GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -113,5 +115,71 @@ private void Delete_Click(object sender, EventArgs e)
                 Close();
             }
         }
+
+        private IReadOnlyList<string> GetAllBranchesWhichContainGivenCommit(ObjectId commitId)
+        {
+            lock (_containedInBranch)
+            {
+                if (_containedInBranch.TryGetValue(commitId, out IReadOnlyList<string> branches))
+                {
+                    return branches;
+                }
+
+                branches = Module.GetAllBranchesWhichContainGivenCommit(commitId, true, false); // include also remotes?
+
+                _containedInBranch.Add(commitId, branches);
+                return branches;
+            }
+        }
+
+        private void BuildContainedInBranchData(IGitRef[] selectedBranches)
+        {
+            if (!selectedBranches.Any())
+            {
+                return;
+            }
+
+            foreach (IGitRef selectedBranch in selectedBranches)
+            {
+                GetAllBranchesWhichContainGivenCommit(selectedBranch.ObjectId);
+            }
+        }
+
+        private void Branches_SelectedValueChanged(object sender, EventArgs e)
+        {
+            ProcessSelectedBranches();
+        }
+
+        private void ProcessSelectedBranches()
+        {
+            IGitRef[] selectedBranches = Branches.GetSelectedBranches().ToArray();
+            CheckSelectedBranches(selectedBranches);
+
+            Task.Run(() =>
+            {
+                BuildContainedInBranchData(selectedBranches);
+            });
+        }
+
+        private void CheckSelectedBranches(IGitRef[] selectedBranches)
+        {
+            if (!selectedBranches.Any())
+            {
+                return;
+            }
+
+            foreach (IGitRef selectedBranch in selectedBranches)
+            {
+                if (!_reflogHashes.Contains(selectedBranch.ObjectId.ToString()))
```

Comment:
This is another O(n^2)...

Reply:
Same than previously because the check result has been cached.